Reg badges

AaronOpfer opened the pull request:
This adds register badges as described in #486 

 * I look for both registers and the values pointed to by registers. The latter might not be very useful but I don't think it hurts anything.
 * If multiple registers point to the same place we append the strings together (badge says `eax, ebx`. this will definitely make us paint over the addresses, though. 
 * They are always white text on blue background. We should probably figure out how we can use the palette to make these badges themeable instead.
 * I changed the line1/line2/line3 resize code so that if the user moves the mouse quickly, instead of ignoring the resize, we resize to the minimum.

10110111 commented:
My concern is that it's not possible to switch this off. As I've tried it, I usually just got some space eaten for nothing, since usually no registers have addresses visible on current page. Only sometimes these badges do appear, but again, IMO they are of not much use.

So I'd rather have an option to turn this off and reclaim the space.

AaronOpfer commented:
I agree that there probably should be an option to disable it. I'll take a look at how difficult that will be.

eteran commented:
Awesome! can you attach a screenshot of how it looks?

AaronOpfer commented:
![image](https://cloud.githubusercontent.com/assets/1189089/21025481/41153b20-bd4e-11e6-9621-9b5e6f0cd150.png)

![image](https://cloud.githubusercontent.com/assets/1189089/21025497/4d219f76-bd4e-11e6-9d52-4e6799cd3cf5.png)

eteran commented:
very cool. If you can make it toggleable as @10110111  suggested, I think it'll be ready to merge :-)

AaronOpfer commented:
I've added a config option. I've also squashed all the commits together.

The only thing I'm not happy about is that I can't listen to "config changed" events to determine if I can move l1, l2 and l3 to the left by 5 characters when users disable, and back again when users enable. I also have to hardcode the 5 character width in two different places, ideally I'd only have to hardcode it in one place. That's just how GUI code ends up going sometimes, though. It's not a big sin.

eteran commented:
I think that I created a configuration changed event at some point. Look to connect to `edb::v1::config()` `SIGNAL(settingsUpdated())`

eteran commented:
This looks good though, we can always improve it with further PRs. Merging!

eteran commented:
@10110111 BTW, just as an explanation of how I think this feature will be useful. I am thinking about how it could help in tracing self modifying code. Like self extracting archives and malware.

Either way, I am very happy with the state of this feature, as it can be toggled to the user's content. Customizing the UI is one of the strengths of edb :-D
